core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -633,6 +634,21 @@ public Result visit(Project e) {
       if ((!isStar(e.getProjects(), e.getInput().getRowType(), e.getRowType())
           || style.isExpandProjection()) && !unpivotRelToSqlUtil.isStarInUnPivot(e, x)) {
         final List<SqlNode> selectList = new ArrayList<>();
+
+        List<String> aliases = new ArrayList<>();
+        if (x.node instanceof SqlSelect) {
+          SqlSelect sqlSelect = (SqlSelect) x.node;
+          if (sqlSelect.getFrom() instanceof SqlPivot) {
+            SqlPivot sqlPivot = (SqlPivot) sqlSelect.getFrom();
+            aliases = sqlPivot.inList.stream()
+                .filter(SqlBasicCall.class::isInstance)
+                .map(SqlBasicCall.class::cast)
+                .filter(basicCall -> basicCall.getOperator() == SqlStdOperatorTable.AS)
+                .filter(basicCall -> basicCall.getOperandList().size() > 1)
+                .map(basicCall -> String.valueOf(basicCall.getOperandList().get(1)))
```

Comment:
Suggestion:

basicCall.operand(1).toString() can be used directly instead of String.valueOf(basicCall.getOperandList().get(1))